### PR TITLE
remove lazy FSharpChecker construction from XmlDocParser and DepthParser

### PR DESCRIPTION
**Pull request by vasily-kirichenko**

### CODE REVIEW

File: src/FSharpVSPowerTools.Core/DepthParser.fs [hunk 1]
```diff
@@ -326,20 +326,11 @@ module internal DepthParsing =
...
-    member internal __.Checker = checker.Value
-
-    static member internal Instance = DepthParser()
 
+type DepthParser (checker: FSharpChecker) =
     /////////////////////////////////////////////////////
-
     // TODO, consider perf, way just to consider viewport and do as little work as necessary?
-
-    member internal x.GetRangesImpl(sourceCodeLinesOfTheFile, sourceCodeOfTheFile, filename, checker: FSharpChecker option) =
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Core/DepthParser.fs]: Make this function `static let` and pass the checker explicitly (do similarly for other `static member`s). I think it makes the intention clear and it is consistent with `XmlDocParser`.

File: src/FSharpVSPowerTools.Logic/DepthTagger.fs [hunk 2]
```diff
@@ -34,8 +34,8 @@ type DepthTagger(buffer: ITextBuffer, filename: string, fsharpLanguageService: V
             let snapshot = buffer.CurrentSnapshot // this is the possibly-out-of-date snapshot everyone here works with
             let sourceCodeOfTheFile = snapshot.GetText()                        
             let! ranges = fsharpLanguageService.CheckerAsync (fun x ->
-                            DepthParser.GetNonoverlappingDepthRanges(sourceCodeOfTheFile, filename, x))
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic/DepthTagger.fs]: Rename `x` to `checker`.

File: src/FSharpVSPowerTools.Logic/XmlDocFilter.fs [hunk 3]
```diff
@@ -43,7 +43,7 @@ type XmlDocFilter(textView: IVsTextView, wpfTextView: IWpfTextView, filename: st
                             // XmlDocable line #1 are 1-based, editor is 0-based
                             let curLineNum = wpfTextView.Caret.Position.BufferPosition.GetContainingLine().LineNumber + 1 
                             let! xmlDocables = languageService.CheckerAsync (fun x -> 
-                                XmlDocParser.GetXmlDocables(wpfTextView.TextSnapshot.GetText(), filename, x))
+                                XmlDocParser.getXmlDocables(wpfTextView.TextSnapshot.GetText(), filename, x))
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic/XmlDocFilter.fs]: Rename `x` to `checker`.

File: src/FSharpVSPowerTools.Core/DepthParser.fs [hunk 4]
```diff
@@ -326,20 +326,11 @@ module internal DepthParsing =
...
+    let getRangesImpl(sourceCodeLinesOfTheFile, sourceCodeOfTheFile, filename) =
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Core/DepthParser.fs]: Make this function static let and pass the checker explicitly (do similarly for other static members). I think it makes the intention clear and it is consistent with XmlDocParser.

**Comment by vasily-kirichenko** [comment]
All done.

**Comment by dungpa** [comment]
Great. Thanks for cleaning it up.

**Pull request merged**